rpcs3/Emu/Cell/SPUThread.cpp

```diff
@@ -388,25 +388,75 @@ void SPUThread::cpu_task()
 		g_cfg.core.spu_decoder == spu_decoder_type::fast ? &g_spu_interpreter_fast.get_table() :
 		(fmt::throw_exception<std::logic_error>("Invalid SPU decoder"), nullptr));
 
-	// LS base address
-	const auto base = vm::_ptr<const u32>(offset);
+	// LS pointer
+	const auto base = vm::_ptr<const u8>(offset);
+	const auto bswap4 = _mm_set_epi8(12, 13, 14, 15, 8, 9, 10, 11, 4, 5, 6, 7, 0, 1, 2, 3);
+
+	v128 _op;
+	using func_t = decltype(&spu_interpreter::UNK);
+	func_t func0, func1, func2, func3, func4, func5;
 
 	while (true)
 	{
-		if (!test(state) || !check_state())
+		if (UNLIKELY(test(state)))
 		{
-			// Read opcode
-			const u32 op = base[pc / 4];
-
-			// Call interpreter function
-			table[spu_decode(op)](*this, { op });
+			if (check_state()) return;
 
-			// Next instruction
-			pc += 4;
+			// Decode single instruction (may be step)
+			const u32 op = *reinterpret_cast<const be_t<u32>*>(base + pc);
```

Comment:
well, if you already put this optimization in your pr, I might close mine

Reply:
I'll wait for testing. It once almost doubled PPU performance, buf I have no idea how it will affect SPU.